node/actors/network/src/http/mod.rs

```diff
@@ -0,0 +1,280 @@
+//! Http Server to export debug information
+use anyhow::{anyhow, Context};
+use base64::Engine;
+use build_html::{Html, HtmlContainer, HtmlPage, Table, TableCell, TableCellType, TableRow};
+use http_body_util::Full;
+use hyper::{
+    body::Bytes,
+    header::{self, HeaderValue},
+    server::conn::http1,
+    service::service_fn,
+    HeaderMap, Request, Response, StatusCode,
+};
+use hyper_util::rt::tokio::TokioIo;
+use im::HashMap;
+use std::{
+    fs, io,
+    net::SocketAddr,
+    path::PathBuf,
+    sync::Arc,
+    time::{Duration, SystemTime},
+};
+use tls_listener::TlsListener;
+use tokio::net::TcpListener;
+use tokio_rustls::{
+    rustls::{
+        pki_types::{CertificateDer, PrivateKeyDer},
+        ServerConfig,
+    },
+    TlsAcceptor,
+};
+use zksync_concurrency::net::http::{DebugCredentials, DebugPageConfig};
+
+use crate::{consensus, ctx, scope, Network, StreamValues};
+
+const STYLE: &str = include_str!("style.css");
+
+/// Http Server.
+pub struct Server {
+    addr: SocketAddr,
+    credentials: Option<DebugCredentials>,
+    cert_path: PathBuf,
+    key_path: PathBuf,
+    network: Arc<Network>,
+}
+
+impl Server {
+    /// Creates a new Server
+    pub fn new(config: DebugPageConfig, network: Arc<Network>) -> Server {
+        Server {
+            addr: config.addr,
```

Comment:
if you just embed the fields of configs, it would be cleaner to have a `config` field in the Server instead.